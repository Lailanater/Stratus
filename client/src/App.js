import React from 'react';
import Wrapper from '../components/Wrapper';
import { Redirect, Route } from 'react-router-dom';
import Header from '../organisms/Header';
import HomePage from './HomePage';
import SettingsPage from './SettingsPage';
import AddProjectPage from './AddProjectPage';
import EditProjectsPage from './EditProjectsPage';
import CreateGrammarPage from './CreateGrammarPage';
import CreateMenuPage from './CreateMenuPage';
import store from '../redux/store';
import { Provider } from 'react-redux';
import { saveState } from '../localStorage';

store.subscribe(() => {
    const currentState = store.getState();
    console.log(currentState);
    saveState(currentState);
});

function App() {
    return (
        <Provider store={store}>
            <Wrapper>
                <Header />
                <Redirect to="/" />
                <Route exact path="/" component={HomePage} />
                <Route path="/settings" component={SettingsPage} />
                <Route path="/addProject" component={AddProjectPage} />
                <Route path="/editProjects" component={EditProjectsPage} />
                <Route path="/createGrammar" component={CreateGrammarPage} />
                <Route path="/createMenu" component={CreateMenuPage} />
            </Wrapper>
        </Provider>
    );
}

export default App;
