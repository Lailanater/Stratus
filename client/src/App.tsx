import React from "react";
import { Home } from "./Home";

interface Props {}

export const App: React.FC<Props> = () => {
  return (
    <div className="vh-100" style={{ backgroundColor: "#303030" }}>
      <Home />
    </div>
  );
};
