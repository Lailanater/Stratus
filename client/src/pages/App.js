import React from 'react';

import { Provider } from "react-redux";
import store from "../redux/store";

import Theming from "./Theming";
import { saveState } from "../localStorage";

function App() {
    store.subscribe(() => {
        const currentState = store.getState();
        console.log(currentState);
        saveState(currentState);
    });

    return (
        <Provider store={store}>
            <Theming />
        </Provider>
    );
}

export default App;
