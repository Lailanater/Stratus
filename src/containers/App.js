import React from 'react';

import {Provider} from "react-redux";
import store from "../redux/store";

import Theming from "../containers/Theming";

function App() {
    store.subscribe(() => console.log(store.getState()));

    return (
        <Provider store={store}>
            <Theming/>
        </Provider>
    );
}

export default App;
