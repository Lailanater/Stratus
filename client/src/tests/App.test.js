import React from 'react';
import renderer from 'react-test-renderer';
import App from '../pages/App';

it('renders without crashing', () => {
  const component = renderer.create(<App />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
