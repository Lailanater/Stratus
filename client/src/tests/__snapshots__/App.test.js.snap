// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
Array [
  <div>
    <header
      className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <button
          aria-label="menu"
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          disabled={false}
          edge="start"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
        <a
          href="/"
          onClick={[Function]}
          style={
            Object {
              "color": "white",
              "textDecoration": "none",
            }
          }
        >
          <h5
            className="MuiTypography-root MuiTypography-h5"
          >
            Stratus
          </h5>
        </a>
      </div>
    </header>
  </div>,
  <div
    className="container"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation5 container MuiPaper-rounded"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated"
      >
        Current Project
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline"
        onClick={[Function]}
      >
        <div
          aria-haspopup="listbox"
          aria-labelledby=" "
          className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </div>
        <input
          onAnimationStart={[Function]}
          type="hidden"
          value=""
        />
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root MuiSelect-icon"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
    <br />
    <div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSizeLarge MuiButton-sizeLarge"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "backgroundColor": "#192a56",
            "color": "white",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Create Menu
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSizeLarge MuiButton-sizeLarge"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "backgroundColor": "#40739e",
            "color": "white",
            "marginLeft": "5px",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Create Grammar
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
]
`;
